File: Trivia/GetRoomStateResponse.cpp
#include "GetRoomStateResponse.h"

GetRoomStateResponse::GetRoomStateResponse(const unsigned int& status, const bool& hasGameBegun, const vector<string> players, const unsigned int& questionCount, const unsigned int& answerTimeout, const unsigned int& maxPlayers) : _status(status), _hasGameBegun(hasGameBegun), _players(players), _questionCount(questionCount), _answerTimeout(answerTimeout), _maxPlayers(maxPlayers)
{}

unsigned int GetRoomStateResponse::getStatus() const
{
	return _status;
}

bool GetRoomStateResponse::getHasGameBegun() const
{
	return _hasGameBegun;
}

void GetRoomStateResponse::startGame()
{
	_hasGameBegun = true;
}

vector<string> GetRoomStateResponse::getPlayers() const
{
	return _players;
}

unsigned int GetRoomStateResponse::getQuestionCount() const
{
	return _questionCount;
}

unsigned int GetRoomStateResponse::getAnswerTimeout() const
{
	return _answerTimeout;
}

unsigned int GetRoomStateResponse::getMaxPlayers() const
{
	return _maxPlayers;
}

File: TriviaClient/Responses/GetRoomStateResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TriviaClient.Responses
{
    class GetRoomStateResponse
    {
        private int status;
        private bool has_game_begun;
        private string[] players;
        private int question_count;
        private int answer_timeout;
        private int max_players;
        public GetRoomStateResponse(int status, bool has_game_begun, string[] players, int question_count, int answer_timeout, int max_players)
        {
            this.status = status;
            this.has_game_begun = has_game_begun;
            this.players = players;
            this.question_count = question_count;
            this.answer_timeout = answer_timeout;
            this.max_players = max_players;
        }

        public int GetStatus()
        {
            return status;
        }

        public bool GetHasGameBegun()
        {
            return has_game_begun;
        }

        public string[] GetPlayers()
        {
            return players;
        }

        public int GetQuestionCount()
        {
            return question_count;
        }

        public int GetAnswerTimeout()
        {
            return answer_timeout;
        }

        public int GetMaxPlayers()
        {
            return max_players;
        }

    }
}
